oauth2_http/javatests/com/google/auth/oauth2/AwsCredentialsTest.java

```diff
@@ -774,6 +862,98 @@ public void credentialSource_missingRegionalCredVerificationUrl() {
+    List<String> regionKeys = ImmutableList.of("AWS_REGION", "AWS_DEFAULT_REGION");
+    for (String regionKey : regionKeys) {
+      TestEnvironmentProvider environmentProvider = new TestEnvironmentProvider();
+      // AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are always required.
+      environmentProvider
+          .setEnv(regionKey, "awsRegion")
+          .setEnv("AWS_ACCESS_KEY_ID", "awsAccessKeyId")
+          .setEnv("AWS_SECRET_ACCESS_KEY", "awsSecretAccessKey");
+      AwsCredentials awsCredential =
+          (AwsCredentials)
+              AwsCredentials.newBuilder(AWS_CREDENTIAL)
+                  .setHttpTransportFactory(transportFactory)
+                  .setCredentialSource(buildAwsImdsv2CredentialSource(transportFactory))
+                  .setEnvironmentProvider(environmentProvider)
+                  .build();
+      assertFalse(awsCredential.shouldUseMetadataServer());
+    }
+  }
+
+  @Test
+  public void shouldUseMetadataServer_missingRegion() {
+    MockExternalAccountCredentialsTransportFactory transportFactory =
+        new MockExternalAccountCredentialsTransportFactory();
+
+    TestEnvironmentProvider environmentProvider = new TestEnvironmentProvider();
+    environmentProvider
+        .setEnv("AWS_ACCESS_KEY_ID", "awsAccessKeyId")
+        .setEnv("AWS_SECRET_ACCESS_KEY", "awsSecretAccessKey");
+    AwsCredentials awsCredential =
+        (AwsCredentials)
+            AwsCredentials.newBuilder(AWS_CREDENTIAL)
+                .setHttpTransportFactory(transportFactory)
+                .setCredentialSource(buildAwsImdsv2CredentialSource(transportFactory))
+                .setEnvironmentProvider(environmentProvider)
+                .build();
+    assertTrue(awsCredential.shouldUseMetadataServer());
+  }
+
+  @Test
+  public void shouldUseMetadataServer_missingAwsAccessKeyId() {
+    MockExternalAccountCredentialsTransportFactory transportFactory =
+        new MockExternalAccountCredentialsTransportFactory();
+
+    // Add required environment variables.
+    List<String> regionKeys = ImmutableList.of("AWS_REGION", "AWS_DEFAULT_REGION");
+    for (String regionKey : regionKeys) {
+      TestEnvironmentProvider environmentProvider = new TestEnvironmentProvider();
+      // AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are always required.
+      environmentProvider
+          .setEnv(regionKey, "awsRegion")
+          .setEnv("AWS_SECRET_ACCESS_KEY", "awsSecretAccessKey");
+      AwsCredentials awsCredential =
+          (AwsCredentials)
+              AwsCredentials.newBuilder(AWS_CREDENTIAL)
+                  .setHttpTransportFactory(transportFactory)
+                  .setCredentialSource(buildAwsImdsv2CredentialSource(transportFactory))
+                  .setEnvironmentProvider(environmentProvider)
+                  .build();
+      assertTrue(awsCredential.shouldUseMetadataServer());
+    }
+  }
+
+  @Test
+  public void shouldUseMetadataServer_missingAwsSecretAccessKey() {
```

Comment:
Consider adding test when both accesskeyid and secretaccesskey are missing. Also when all 4 env vars are missing.